### PR TITLE
Split repo into two and fixups

### PR DESCRIPTION
**Pull request by Morganamilo**
Splits the repo into this -src repo and a repo that contains a generated .js file.

Also fixes up most of the non code writing issue in https://github.com/ably-labs/test-observability-action/issues

The publish workflow seems to work but needs a secret added to actually push to the repo.

The test CI has been removed for now as it ran against a fake server won't work with the repo split. A way to test this action needs to be figured out.

### CODE REVIEW

**Comment by Morganamilo** [comment]
I believe you can run actions via a file path. So in our CI check we could probably run `npm install` then continue as usual.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        "browser": true,
+        "node": true,
+        "commonjs": true,
+        "es2021": true
+    },
+    "extends": "eslint:recommended",
+    "parserOptions": {
+        "ecmaVersion": "latest"
+    },
+    "rules": {
```
Comment (owenpearson) [inline .eslintrc.js]: We should probably just remove this field if it's empty

**Review by owenpearson — CHANGES_REQUESTED** [review]
Looks great. Just one minor nitpick.

File: .eslintrc.js (hunk 1 above)
Comment (Morganamilo) [reply .eslintrc.js]: This was auto generated by  `npm init @eslint/config` so I don't really feel like like editing it. Gonna merge now to speed things up. Let me know if you have a strong opinion.

**Pull request merged**